File: components/permissions/src/test/java/org/cloudfoundry/credhub/services/TestPasswordKeyProxyFactory.java
package org.cloudfoundry.credhub.services;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

import org.cloudfoundry.credhub.config.EncryptionKeyMetadata;

@Service
@SuppressWarnings("unused")
@Profile("unit-test")
public class TestPasswordKeyProxyFactory implements PasswordKeyProxyFactory {
  @Override
  public KeyProxy createPasswordKeyProxy(
    final EncryptionKeyMetadata encryptionKeyMetadata, final InternalEncryptionService encryptionService) {
    return new PasswordBasedKeyProxy(encryptionKeyMetadata.getEncryptionPassword(), 1, encryptionService);
  }
}

File: components/permissions/src/test/kotlin/org/cloudfoundry/credhub/services/TestPasswordKeyProxyFactory.kt
package org.cloudfoundry.credhub.services

import org.cloudfoundry.credhub.config.EncryptionKeyMetadata
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service

@Service
@Profile("unit-test")
class TestPasswordKeyProxyFactory : PasswordKeyProxyFactory {
    override fun createPasswordKeyProxy(
        encryptionKeyMetadata: EncryptionKeyMetadata,
        encryptionService: InternalEncryptionService,
    ): KeyProxy {
        return PasswordBasedKeyProxy(encryptionKeyMetadata.encryptionPassword, 1, encryptionService)
    }
}
